src/magnet_uri.cpp

```diff
@@ -258,14 +258,68 @@ namespace libtorrent {
 			return p;
 		}
 
+		auto select_pos = std::string::npos;
+		string_view select = url_has_argument(uri, "so", &select_pos);
+		while (!select.empty())
+		{
+			// parse the ranges or indices
+			do
+			{
+				string_view token;
+				std::tie(token, select) = split_string(select, ',');
+
+				int idx1, idx2;
+				// TODO: what's the right number here?
+				constexpr int max_index = 10000; // can't risk out of memory
+
+				auto const divider = token.find_first_of('-');
+				if (divider != std::string::npos) // it's a range
+				{
+					if (divider == 0) // no start index
```

Comment:
ideally there would be a test case for every failure that's handled here

Reply:
including indices > 10000

Reply:
OK